### PR TITLE
Add the verbose option

### LINKED ISSUE #7: A verbose option?
@sindresorhus 
In places where a verbose display is required, I would like something like : -
```js
prettyMs(1337, {verbose: true});
// => 1.3 seconds
// => instead of `1.3s`

prettyMs(1337000000, {verbose: true});
//=> '15 days 11 hours 23 minutes 20 seconds'
```

I would be happy to submit a patch, if you are interested in this too?

### PR DESCRIPTION
**Pull request by icyflame**
Fixes #7

I have also added the option to the README.

***

/ @sindresorhus

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 0}), '33s');
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 4}), '33.3330s');
 });
+
+it('should have a verbose option', function(){
+
+	var use_verbose = function(ms){
+		return prettyMs(ms, {verbose: true});
+	}
```
Comment (kevva) [inline test.js]: Use the `function` syntax instead.

```js
function prettyMsVerbose(ms) {
    return prettyMs(ms, {verbose: true});
}
```

File: index.js [hunk 2]
```diff
@@ -14,7 +14,13 @@ module.exports = function (ms, opts) {
 		throw new TypeError('Expected a number');
 	}
 
+	var verbose = opts ? opts.verbose : false;
```
Comment (kevva) [inline index.js]: Remove this and just check for `opts.verbose` instead.

File: index.js [hunk 3]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
 	var ret = [];
 	var parsed = parseMs(ms);
 
-	ret = add(ret, parsed.days, 'd');
-	ret = add(ret, parsed.hours, 'h');
-	ret = add(ret, parsed.minutes, 'm');
+	if(verbose){
```
Comment (kevva) [inline index.js]: `if (opts.verbose) {`, notice the spacing too.

File: index.js [hunk 4]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
...
+		ret = add(ret, parsed.days, parsed.days == 1 ? ' day' : ' days');
+		ret = add(ret, parsed.hours, parsed.hours == 1 ? ' hour' : ' hours');
+		ret = add(ret, parsed.minutes, parsed.minutes == 1 ? ' minute' : ' minutes');
+	}
+	else{
```
Comment (kevva) [inline index.js]: `} else {`

File: index.js [hunk 5]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
...
+		ret = add(ret, parsed.days, 'd');
+		ret = add(ret, parsed.hours, 'h');
+		ret = add(ret, parsed.minutes, 'm');
+	}
 
 	if (opts.compact) {
-		ret = add(ret, parsed.seconds, 's');
+		if(verbose)
```
Comment (kevva) [inline index.js]: `if (opts.verbose) {`, don't use shorthand `if` statements.

File: index.js [hunk 6]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
...
+			ret = add(ret, parsed.seconds, parsed.seconds === 1 ? ' second' : ' seconds');
+		else
```
Comment (kevva) [inline index.js]: `} else {`

File: index.js [hunk 7]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
...
+			ret = add(ret, parsed.seconds, 's');
 		return '~' + ret[0];
 	}
 
-	ret = add(ret, (ms / 1000 % 60).toFixed(secDecimalDigits).replace(/\.0$/, ''), 's');
+	var seconds_fixed = (ms / 1000 % 60).toFixed(secDecimalDigits);
```
Comment (kevva) [inline index.js]: Use camelCase.

File: index.js [hunk 8]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
...
+	
+	if(verbose)
```
Comment (kevva) [inline index.js]: `if (opts.verbose) {`, no shorthand `if` statements.

File: readme.md [hunk 9]
```diff
@@ -55,6 +55,12 @@ Type: `boolean`
 
 Only show the first unit: `1h 10m` ➔ `~1h`.
 
+##### verbose
+
+Type: `boolean`
```
Comment (arthurvr) [inline readme.md]: This misses the `Default`

File: test.js [hunk 10]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	var use_decimal_digits = function(ms){
+		return prettyMs(ms, { verbose: true, secDecimalDigits: 4 });
+	}
+
+	assert.strictEqual(use_decimal_digits(1000), '1.0000 second');
+	assert.strictEqual(use_decimal_digits(1000 + 400), '1.4000 seconds');
+	assert.strictEqual(use_decimal_digits(1000 * 2 + 400), '2.4000 seconds');
+	assert.strictEqual(use_decimal_digits(1000 * 5 + 254), '5.2540 seconds');
+	assert.strictEqual(use_decimal_digits(33333), '33.3330 seconds');
+});
```
Comment (arthurvr) [inline test.js]: missing newline

File: test.js [hunk 11]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 0}), '33s');
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 4}), '33.3330s');
 });
+
+it('should have a verbose option', function(){
```
Comment (arthurvr) [inline test.js]: spacing `function () {`

File: test.js [hunk 12]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	assert.strictEqual(use_verbose(1000 * 67), '1 minute 7 seconds');
+	assert.strictEqual(use_verbose(1000 * 60 * 5), '5 minutes');
+	assert.strictEqual(use_verbose(1000 * 60 * 67), '1 hour 7 minutes');
+	assert.strictEqual(use_verbose(1000 * 60 * 60 * 12), '12 hours');
+	assert.strictEqual(use_verbose(1000 * 60 * 60 * 40), '1 day 16 hours');
+	assert.strictEqual(use_verbose(1000 * 60 * 60 * 999), '41 days 15 hours');
+
+});
+
+it('should work with verbose and compact options', function(){
```
Comment (arthurvr) [inline test.js]: spacing `function () {`

File: test.js [hunk 13]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	assert.strictEqual(use_verbose_compact(1000 * 67), '~1 minute');
+	assert.strictEqual(use_verbose_compact(1000 * 60 * 5), '~5 minutes');
+	assert.strictEqual(use_verbose_compact(1000 * 60 * 67), '~1 hour');
+	assert.strictEqual(use_verbose_compact(1000 * 60 * 60 * 12), '~12 hours');
+	assert.strictEqual(use_verbose_compact(1000 * 60 * 60 * 40), '~1 day');
+	assert.strictEqual(use_verbose_compact(1000 * 60 * 60 * 999), '~41 days');
+
+});
+
+it('should work with verbose and secDecimalDigits options', function(){
```
Comment (arthurvr) [inline test.js]: spacing `function () {`

File: index.js [hunk 14]
```diff
@@ -14,7 +14,13 @@ module.exports = function (ms, opts) {
 		throw new TypeError('Expected a number');
 	}
 
+	var verbose = opts ? opts.verbose : false;
+
 	if (ms < 1000) {
+		if(verbose && Math.ceil(ms) == 1.0)
```
Comment (kevva) [inline index.js]: `if (opts.verbose && Math.ceil(ms) == 1.0) {`

File: test.js (hunk 13 above)
Comment (icyflame) [inline test.js]: @arthurvr Thank you! I have noted the changes requested by @kevva and am changing my whole code according to this! (Spacing, shorthand if, variable naming conventions, function syntax)

File: index.js [hunk 15]
```diff
@@ -14,7 +14,13 @@ module.exports = function (ms, opts) {
...
+			return Math.ceil(ms) + ' millisecond';
+		if(verbose)
```
Comment (kevva) [inline index.js]: `if (opts.verbose) {`

File: index.js (hunk 14 above)
Comment (arthurvr) [inline index.js]: also use a triple equal where possible

File: index.js [hunk 16]
```diff
@@ -24,16 +30,31 @@ module.exports = function (ms, opts) {
 	var ret = [];
 	var parsed = parseMs(ms);
 
-	ret = add(ret, parsed.days, 'd');
-	ret = add(ret, parsed.hours, 'h');
-	ret = add(ret, parsed.minutes, 'm');
+	if(verbose){
+		ret = add(ret, parsed.days, parsed.days == 1 ? ' day' : ' days');
+		ret = add(ret, parsed.hours, parsed.hours == 1 ? ' hour' : ' hours');
+		ret = add(ret, parsed.minutes, parsed.minutes == 1 ? ' minute' : ' minutes');
```
Comment (kevva) [inline index.js]: `===`

File: readme.md (hunk 9 above)
Comment (icyflame) [inline readme.md]: I have added the `Default` option to both `verbose` and `compact`, it was missing earlier!

**Comment by icyflame** [comment]
@kevva 

I have made all of the changes requested : -

- remove shorthand if
- use === wherever possible
- camelCase variable names
- use `opts.verbose` instead of `verbose`
- add `default` options to the readme file

Can you please point me to the style guide that you use for this project, for future reference?

***

cc @arthurvr @sindresorhus

File: index.js [hunk 17]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
 		throw new TypeError('Expected a number');
 	}
 
+	opts = opts || {};
+
 	if (ms < 1000) {
+		if(opts.verbose && Math.ceil(ms) === 1.0) {
```
Comment (kevva) [inline index.js]: Fix spacing. `if (opts.verbose && Math.ceil(ms) === 1.0) {`

File: index.js [hunk 18]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
+			return Math.ceil(ms) + ' millisecond';
+		}
+		if(opts.verbose) {
```
Comment (kevva) [inline index.js]: Fix spacing here too `if (opts.verbose) {`

File: index.js [hunk 19]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
-	opts = opts || {};
-
 	var secDecimalDigits = typeof opts.secDecimalDigits === 'number' ? opts.secDecimalDigits : 1;
 	var ret = [];
 	var parsed = parseMs(ms);
 
-	ret = add(ret, parsed.days, 'd');
-	ret = add(ret, parsed.hours, 'h');
-	ret = add(ret, parsed.minutes, 'm');
+	if(opts.verbose) {
```
Comment (kevva) [inline index.js]: Fix spacing here too `if (opts.verbose) {`.

File: index.js [hunk 20]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
+		ret = add(ret, parsed.minutes, parsed.minutes === 1 ? ' minute' : ' minutes');
+	} else {
+		ret = add(ret, parsed.days, 'd');
+		ret = add(ret, parsed.hours, 'h');
+		ret = add(ret, parsed.minutes, 'm');
+	}
 
 	if (opts.compact) {
-		ret = add(ret, parsed.seconds, 's');
+		if(opts.verbose) {
```
Comment (kevva) [inline index.js]: Fix spacing here too `if (opts.verbose) {`.

File: index.js [hunk 21]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
+		} else {
+			ret = add(ret, parsed.seconds, 's');
+		}
 		return '~' + ret[0];
 	}
 
-	ret = add(ret, (ms / 1000 % 60).toFixed(secDecimalDigits).replace(/\.0$/, ''), 's');
+	var secondsFixed = (ms / 1000 % 60).toFixed(secDecimalDigits);
+	
+	if(opts.verbose) {
```
Comment (kevva) [inline index.js]: Fix spacing here too `if (opts.verbose) {`.

File: test.js [hunk 22]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 0}), '33s');
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 4}), '33.3330s');
 });
+
+it('should have a verbose option', function () {
+
+	function prettyMsVerbose (ms) {
```
Comment (icyflame) [inline test.js]: @kevva Is the spacing okay in this line?

**Comment by icyflame** [comment]
@kevva Done! Sorry, I missed the space between `if` and `(condition)` in the earlier comments. There were also some issues in the function spacing, as pointed out by @arthurvr , which I have fixed too.

File: test.js [hunk 23]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 67), '~1 minute');
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 60 * 5), '~5 minutes');
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 60 * 67), '~1 hour');
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 60 * 60 * 12), '~12 hours');
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 60 * 60 * 40), '~1 day');
+	assert.strictEqual(prettyMsVerboseCompact(1000 * 60 * 60 * 999), '~41 days');
+
+});
+
+it('should work with verbose and secDecimalDigits options', function() {
```
Comment (arthurvr) [inline test.js]: `function () {`

File: test.js [hunk 24]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+
+	function prettyMsVerboseDecimalDigits (ms) {
```
Comment (arthurvr) [inline test.js]: `function prettyMsVerboseDecimalDigits(ms) {`

File: test.js [hunk 25]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+		return prettyMs(ms, { verbose: true, secDecimalDigits: 4 });
```
Comment (arthurvr) [inline test.js]: `{verbose: true, secDecimalDigits: 4}`

File: test.js [hunk 26]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 67), '1 hour 7 minutes');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 12), '12 hours');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 40), '1 day 16 hours');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 999), '41 days 15 hours');
+
+});
+
+it('should work with verbose and compact options', function() {
+
+	function prettyMsVerboseCompact (ms) {
```
Comment (arthurvr) [inline test.js]: `function prettyMsVerboseCompact(ms) {`

File: test.js [hunk 27]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
...
+	assert.strictEqual(prettyMsVerbose(1000 * 67), '1 minute 7 seconds');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 5), '5 minutes');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 67), '1 hour 7 minutes');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 12), '12 hours');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 40), '1 day 16 hours');
+	assert.strictEqual(prettyMsVerbose(1000 * 60 * 60 * 999), '41 days 15 hours');
+
+});
+
+it('should work with verbose and compact options', function() {
```
Comment (arthurvr) [inline test.js]: `function () {`

File: test.js (hunk 22 above)
Comment (arthurvr) [inline test.js]: No it isn't: `function prettyMsVerbose(ms) {`. You only need a space after the `function` keyword.

**Comment by icyflame** [comment]
@kevva More spacing changes!

I have followed this convention (mentioned by @arthurvr ) **a space after the `function` keyword, and a space before the opening curly brace.**

File: index.js [hunk 28]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
 
+	opts = opts || {};
+
 	if (ms < 1000) {
+		if (opts.verbose && Math.ceil(ms) === 1.0) {
+			return Math.ceil(ms) + ' millisecond';
+		}
+		if (opts.verbose) {
+			return Math.ceil(ms) + ' milliseconds';
+		}
```
Comment (sindresorhus) [inline index.js]: This can be simplified with a ternary.

File: index.js [hunk 29]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
-
 	var secDecimalDigits = typeof opts.secDecimalDigits === 'number' ? opts.secDecimalDigits : 1;
 	var ret = [];
 	var parsed = parseMs(ms);
 
-	ret = add(ret, parsed.days, 'd');
-	ret = add(ret, parsed.hours, 'h');
-	ret = add(ret, parsed.minutes, 'm');
+	if (opts.verbose) {
+		ret = add(ret, parsed.days, parsed.days === 1 ? ' day' : ' days');
```
Comment (sindresorhus) [inline index.js]: All of these could just use a naive pluralize function:

```js
var pluralize = function (str, count) {
  return str === 1 ? str : str + 's';
};
```

Comment (sindresorhus) [inline index.js]: Just use https://github.com/sindresorhus/plur for all of the pluralizations.

File: index.js [hunk 30]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
 		return '~' + ret[0];
 	}
 
-	ret = add(ret, (ms / 1000 % 60).toFixed(secDecimalDigits).replace(/\.0$/, ''), 's');
+	var secondsFixed = (ms / 1000 % 60).toFixed(secDecimalDigits);
+	
+	if (opts.verbose) {
+		ret = add(ret, secondsFixed.replace(/\.0$/, ''), secondsFixed === (1).toFixed(secDecimalDigits) ? ' second' : ' seconds');
+	} else {
+		ret = add(ret, secondsFixed.replace(/\.0$/, ''), 's');
```
Comment (sindresorhus) [inline index.js]: `secondsFixed.replace(/\.0$/, '')` duplicated twice here.

File: index.js [hunk 31]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
+			ret = add(ret, parsed.seconds, 's');
+		}
 		return '~' + ret[0];
 	}
 
-	ret = add(ret, (ms / 1000 % 60).toFixed(secDecimalDigits).replace(/\.0$/, ''), 's');
+	var secondsFixed = (ms / 1000 % 60).toFixed(secDecimalDigits);
+	
+	if (opts.verbose) {
+		ret = add(ret, secondsFixed.replace(/\.0$/, ''), secondsFixed === (1).toFixed(secDecimalDigits) ? ' second' : ' seconds');
```
Comment (sindresorhus) [inline index.js]: Just do a `var` for the vanilla seconds and you won't have to do a `(1).toFixed(secDecimalDigits)` here.

File: test.js [hunk 32]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 0}), '33s');
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 4}), '33.3330s');
 });
+
+it('should have a verbose option', function () {
+
+	function prettyMsVerbose(ms) {
```
Comment (sindresorhus) [inline test.js]: Can just be named `fn` as it looks nicer.

File: index.js [hunk 33]
```diff
@@ -14,26 +14,48 @@ module.exports = function (ms, opts) {
...
+		ret = add(ret, parsed.minutes, 'm');
+	}
 
 	if (opts.compact) {
-		ret = add(ret, parsed.seconds, 's');
+		if (opts.verbose) {
+			ret = add(ret, parsed.seconds, parsed.seconds === 1 ? ' second' : ' seconds');
+		} else {
+			ret = add(ret, parsed.seconds, 's');
+		}
```
Comment (sindresorhus) [inline index.js]: Ternary

File: readme.md [hunk 34]
```diff
@@ -52,9 +52,17 @@ Number of digits to appear after the seconds decimal point.
 ##### compact
 
 Type: `boolean`
```
Comment (sindresorhus) [inline readme.md]: Needs to spaces at the end to make a markdown linebreak.

File: readme.md (hunk 9 above)
Comment (sindresorhus) [inline readme.md]: https://github.com/sindresorhus/pretty-ms/pull/8/files#r32889399 here too.

File: cli.js [hunk 35]
```diff
@@ -21,7 +21,8 @@ var cli = meow({
 
 function init(data) {
 	console.log(prettyMs(Number(data), {
-		compact: cli.flags.compact
+		compact: cli.flags.compact,
+		verbose: cli.flags.verbose
```
Comment (sindresorhus) [inline cli.js]: Needs to be added to the readme too.

File: index.js (hunk 31 above)
Comment (icyflame) [inline index.js]: @sindresorhus I am sorry, I don't understand you. What do you mean `vanilla seconds` ?

File: readme.md (hunk 34 above)
Comment (icyflame) [inline readme.md]: @sindresorhus This is how it renders right now : -

image

What do you mean by `needs to spaces at the end` ?

File: index.js (hunk 31 above)
Comment (sindresorhus) [inline index.js]:
```js
var seconds = ms / 1000 % 60;
var secondsFixed = seconds.toFixed(secDecimalDigits);
```

Then you can use `seconds` directly in the above.

File: readme.md (hunk 34 above)
Comment (sindresorhus) [inline readme.md]: And do you see a difference between how the `secDecimalDigits` content looks compared to `verbose` and `compact`?

> What do you mean by needs to spaces at the end ?

Exactly what I said. 2 spaces is how you make a linebreak in markdown.

File: index.js [hunk 36]
```diff
@@ -14,26 +14,34 @@ module.exports = function (ms, opts) {
 		throw new TypeError('Expected a number');
 	}
 
+	opts = opts || {};
+	var plur = require('plur');
```
Comment (arthurvr) [inline index.js]: require this at the top of the file

File: test.js [hunk 37]
```diff
@@ -27,3 +27,59 @@ it('should have a secDecimalDigits option', function () {
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 0}), '33s');
 	assert.strictEqual(prettyMs(33333, {secDecimalDigits: 4}), '33.3330s');
 });
+
+it('should have a verbose option', function () {
+
```
Comment (kevva) [inline test.js]: The empty lines in the beginning and end of every `it` should be removed to.

```js
it('should...', function () {
    [code]
});
```

File: readme.md (hunk 34 above)
Comment (icyflame) [inline readme.md]: Got it! :+1: Thanks!

**Comment by icyflame** [comment]
@kevva Done! There were empty lines before `it`, in the existing code. I have removed them, as well as existing empty lines.

@sindresorhus everything has been changed to use `plur`. I also made the changes to `readme`.

**Pull request closed without merging**